gcs/smart_entry.hh

```diff
@@ -0,0 +1,130 @@
+#ifndef GLASGOW_CONSTRAINT_SOLVER_GUARD_GCS_SMART_ENTRY_HH
+#define GLASGOW_CONSTRAINT_SOLVER_GUARD_GCS_SMART_ENTRY_HH
+
+#include <variant>
+#include <vector>
+#include <gcs/variable_id.hh>
+#include <gcs/integer.hh>
+#include <algorithm> //std::sort
+using namespace gcs;
+
+enum ConstraintType {
+    LESS_THAN,
+    LESS_THAN_EQUAL,
+    EQUAL,
+    NOT_EQUAL,
+    GREATER_THAN,
+    GREATER_THAN_EQUAL,
+    IN,
+    NOT_IN
+};
+
+/**
+ * \brief Smart entries for smart tables
+ */
+struct BinaryEntry {
+    IntegerVariableID var_1;
+    IntegerVariableID var_2;
+    ConstraintType constraint_type;
+
+    // Constructor
```

Comment:
Your ctor is what would be created by default, you don't need to specify it.

Reply:
For some reason I get an error with the child structures if I don't have a constructor

`No matching constructor for initialization of 'BinaryEntry' candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 3 were provided candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 3 were provided`